File: schneider_kalf/src/form/nodes/expressions/BinaryExpression.ts
import Expression from "./Expression";
import TreeNode from "../TreeNode";
import ExpressionVisitor from "../visitors/ExpressionVisitor";

export default abstract class BinaryExpression implements TreeNode {
  get right(): Expression {
    return this._right;
  }

  get left(): Expression {
    return this._left;
  }

  private _left: Expression;
  private _right: Expression;

  constructor(left: Expression, right: Expression) {
    this._left = left;
    this._right = right;
  }

  abstract accept(visitor: ExpressionVisitor): any;
}

File: Remi_Hasan/src/main/java/ql/model/expression/binary/BinaryExpression.java
package ql.model.expression.binary;

import ql.model.expression.Expression;

public abstract class BinaryExpression extends Expression {
    private final Expression left;
    private final Expression right;

    protected BinaryExpression(Expression left, Expression right) {
        this.left = left;
        this.right = right;
    }

    public Expression getLeft() {
        return left;
    }

    public Expression getRight() {
        return right;
    }
}